src/Razor/src/Microsoft.VisualStudio.LanguageServices.Razor/Debugging/RazorLanguageService`IVsLanguageDebugInfo.cs

```diff
@@ -48,20 +51,35 @@ internal partial class RazorLanguageService : IVsLanguageDebugInfo
             throw new ArgumentNullException(nameof(editorAdaptersFactory));
         }
 
+        if (lspServerActivationTracker is null)
+        {
+            throw new ArgumentNullException(nameof(lspServerActivationTracker));
+        }
+
         if (joinableTaskFactory is null)
         {
             throw new ArgumentNullException(nameof(joinableTaskFactory));
         }
 
         _breakpointResolver = breakpointResolver;
         _proximityExpressionResolver = proximityExpressionResolver;
+        _lspServerActivationTracker = lspServerActivationTracker;
         _uiThreadOperationExecutor = uiThreadOperationExecutor;
         _editorAdaptersFactory = editorAdaptersFactory;
         _joinableTaskFactory = joinableTaskFactory;
     }
 
     public int GetProximityExpressions(IVsTextBuffer pBuffer, int iLine, int iCol, int cLines, out IVsEnumBSTR? ppEnum)
     {
+        if (!_lspServerActivationTracker.IsActive)
+        {
+            // We can't do anything if our LSP server isn't up and running, and can't initialize here due to UI thread dependency issues
+            // This method should only be called during a debugging sessions, so we should never hit this case, and if we do, we have much
+            // bigger problems.
+            ppEnum = null;
+            return VSConstants.E_NOTIMPL;
```

Comment:
Is there a more correct value (HRESULT) to return here?

Reply:
The callers don't make any interesting distinction between failure results:

https://devdiv.visualstudio.com/DevDiv/_git/VS?path=/src/debugger/vsdebug/package/dbgauto.cpp&version=GBmain&line=373&lineEnd=374&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents
https://devdiv.visualstudio.com/DevDiv/_git/VS?path=/src/debugger/vsdebug/package/dbgtxt.cpp&version=GBmain&line=1376&lineEnd=1377&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents